chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/curesStatistics/email/CuresStatisticsChartSpreadsheet.java

```diff
@@ -0,0 +1,299 @@
+import lombok.extern.log4j.Log4j2;
+
+@Component
+@Log4j2
+public class CuresStatisticsChartSpreadsheet {
+    private static final Integer EXISTING_CERTIFICATION_COL_IDX = 1;
+    private static final Integer NEW_CERTIFICATION_COL_IDX = 2;
+    private static final Integer REQUIRES_UPDATE_COL_IDX = 3;
+    private static final Integer LISTING_COUNT_COL_IDX = 4;
+    private static final Integer PERCENT_CURES_COL_IDX = 6;
+
+    private static final Integer B_1_CURES_ROW_IDX = 1;
+    private static final Integer B_2_CURES_ROW_IDX = 2;
+    private static final Integer B_3_CURES_ROW_IDX = 3;
+    private static final Integer B_7_CURES_ROW_IDX = 4;
+    private static final Integer B_8_CURES_ROW_IDX = 5;
+    private static final Integer B_9_CURES_ROW_IDX = 6;
+    private static final Integer B_10_ROW_IDX = 7;
+    private static final Integer C_3_CURES_ROW_IDX = 8;
+    private static final Integer D_2_CURES_ROW_IDX = 9;
+    private static final Integer D_3_CURES_ROW_IDX = 10;
+    private static final Integer D_10_CURES_ROW_IDX = 11;
+    private static final Integer D_12_ROW_IDX = 12;
+    private static final Integer D_13_ROW_IDX = 13;
+    private static final Integer E_1_CURES_ROW_IDX = 14;
+    private static final Integer F_5_CURES_ROW_IDX = 15;
+    private static final Integer G_6_CURES_ROW_IDX = 16;
+    private static final Integer G_9_CURES_ROW_IDX = 17;
+    private static final Integer G_10_ROW_IDX = 18;
+
+    @Value("${curesStatisticsChartSpreadsheetTemplate}")
+    private String template;
+
+    @Value("${downloadFolderPath}")
+    private String downloadPath;
+
+    private CertificationCriterionService criterionService;
+
+    private List<CriteraToRowMap> criteriaToRowMaps = new ArrayList<CuresStatisticsChartSpreadsheet.CriteraToRowMap>();
+
+    @Autowired
+    public CuresStatisticsChartSpreadsheet(CertificationCriterionService criterionService) {
+        this.criterionService = criterionService;
+
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_1_CURES, B_1_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_2_CURES, B_2_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_3_CURES, B_3_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_7_CURES, B_7_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_8_CURES, B_8_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_9_CURES, B_9_CURES_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.B_10, B_10_ROW_IDX));
+        criteriaToRowMaps.add(new CriteraToRowMap(CertificationCriterionService.Criteria2015.C_3_CURES, C_3_CURES_ROW_IDX));
```

Comment:
It's odd, I know we try to not use magic numbers, but I've always had the feeling that I just didn't know why it mattered. Now I get it. This block of code is _so_ much easier to parse than it would be with a bunch of integers in these calls. 

I _knew_ removing magic numbers was a good idea because I was told it was, but this part demonstrated it in a way that now I feel it :)